### PR TITLE
feat(saved-searches): Use Saved Searches store in OrganizationStream

### PR DESCRIPTION
**Pull request by billyvg**
Use store in org stream view

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationStream/overview.jsx [hunk 1]
```diff
@@ -122,7 +125,10 @@ const OrganizationStream = createReactClass({
     const newQuery = this.props.location.query;
 
     // Wait for saved searches to load before we attempt to fetch stream data
-    if (this.state.savedSearchLoading) {
+    if (this.props.savedSearchLoading) {
+      return;
+    } else if (prevProps.savedSearchLoading) {
+      this.fetchData();
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: Could we hit this path multiple times before the issue list comes down? Should this also include `this.props.savedSearchLoading === false && prevProps.savedSearchLoading)`?

File: src/sentry/static/sentry/app/views/organizationStream/overview.jsx [hunk 2]
```diff
@@ -370,16 +379,16 @@ const OrganizationStream = createReactClass({
...
       this.fetchData();
     } else {
-      this.transitionTo({query});
+      this.transitionTo(null, {query});
     }
   },
 
   onSortChange(sort) {
-    this.transitionTo({sort});
+    this.transitionTo(null, {sort});
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: Should changing sort directions remove a saved search?

File: src/sentry/static/sentry/app/actionCreators/savedSearches.jsx [hunk 3]
```diff
@@ -21,7 +21,13 @@ export function fetchSavedSearches(api, orgId, useOrgSavedSearches = false) {
 
   promise
     .then(resp => {
-      SavedSearchesActions.fetchSavedSearchesSuccess(resp);
+      // Add in project slugs so that we can display them in the picker bars.
+      const savedSearchList = resp.map(search => ({
+        ...search,
+        projectSlug: projectMap[search.projectId],
+      }));
```
Comment (markstory) [inline src/sentry/static/sentry/app/actionCreators/savedSearches.jsx]: Would it be good to leave a TODO to remove this code when org-saved-searches is mainlined?

File: tests/js/spec/views/organizationStream/overview.spec.jsx [hunk 4]
```diff
@@ -95,17 +108,309 @@ describe('OrganizationStream', function() {
...
+      // Loading saved searches
+      expect(savedSearchesRequest).toHaveBeenCalledTimes(1);
+      // Update stores with saved searches
+      await tick();
+      wrapper.update();
+
+      expect(recentSearchesRequest).toHaveBeenCalledTimes(1);
+
+      // Check for fetch tags req
+      // Check for fetch member list req
```
Comment (markstory) [inline tests/js/spec/views/organizationStream/overview.spec.jsx]: Are these todos?

File: src/sentry/static/sentry/app/views/organizationStream/overview.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: Hmm that's what the `else if` checks -- I can make it more explicit if it makes it read more clearly.

Comment (markstory) [reply src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: I'm a muppet, carry on.

File: src/sentry/static/sentry/app/views/organizationStream/overview.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationStream/overview.jsx]: Fixed + added a test

**Review by markstory — APPROVED** [review]

**Pull request merged**